packages/react/src/exchanges/hooks/useExchangeFilters.ts

```diff
@@ -0,0 +1,111 @@
+  type StoreState,
+} from '@farfetch/blackout-redux';
+import { useCallback, useMemo } from 'react';
+import { useSelector } from 'react-redux';
+import useAction from '../../helpers/useAction.js';
+import type {
+  Config,
+  ExchangeFilterItem,
+  OrderItem,
+  PostExchangeFilterData,
+} from '@farfetch/blackout-client';
+
+type PostExchangeFilterDataHook = PostExchangeFilterData & {
+  exchangeFilterItems: Array<ExchangeFilterItemWithOptionalUuid>;
+};
+
+type ExchangeFilterItemWithOptionalUuid = Omit<
+  ExchangeFilterItem,
+  'orderItemUuid'
+> & {
+  orderItemUuid?: OrderItem['shippingOrderLineId'];
+};
+
+/**
+ * Provides facilities to create an exchange filter.
+ */
+function useExchangeFilters(orderItemUuid?: OrderItem['shippingOrderLineId']) {
+  const orderItemUuidHookParameter = orderItemUuid;
+  const isCreating = useSelector((state: StoreState) =>
+    orderItemUuidHookParameter
+      ? isExchangeFilterLoading(state, orderItemUuidHookParameter)
+      : false,
+  );
+  const error = useSelector((state: StoreState) =>
+    orderItemUuidHookParameter
+      ? getExchangeFilterError(state, orderItemUuidHookParameter)
+      : null,
+  );
+  const exchangeFilters = useSelector(getExchangeFilters);
+  const createExchangeFilter = useAction(createExchangeFilterAction);
+  const reset = useAction(resetExchangeFilters);
+
+  /**
+   * Creates an exchange filter. This method will save the exchange filter by orderItemUuid without
+   * replacing existing filters. Will throw if no exchangeFilterData is passed
+   * to create the filter.
+   *
+   * @param exchangeFilterData - Data to create the exchange filter.
+   * @param config - Custom configurations to send to the client instance (axios).
+   *
+   * @returns Promise that will be resolved when the call to the endpoint finishes.
+   */
+  const create = useCallback(
+    (exchangeFilterData: PostExchangeFilterDataHook, config?: Config) => {
+      if (!exchangeFilterData) {
+        return Promise.reject(new Error('No data to filter provided'));
+      }
+
+      // Although exchangeFilterItems is an array, it only has 1 object at a time, so it's safe to
+      // directly access the first index.
+      const orderItemUuid =
+        exchangeFilterData.exchangeFilterItems[0]?.orderItemUuid;
+
+      if (!orderItemUuid && !orderItemUuidHookParameter) {
+        return Promise.reject(
+          new Error(
+            'No orderItemUuid provided to either the hook or the create function',
+          ),
+        );
+      }
+
+      exchangeFilterData.exchangeFilterItems.forEach(item => {
+        if (
+          !item.hasOwnProperty('orderItemUuid') &&
+          orderItemUuidHookParameter
+        ) {
+          item.orderItemUuid = orderItemUuidHookParameter;
+        }
+      });
+
+      return createExchangeFilter(exchangeFilterData, config);
+    },
+    [createExchangeFilter, orderItemUuidHookParameter],
+  );
+
+  const data = useMemo(() => {
+    if (!exchangeFilters) {
```

Comment:
This is unnecessary, you can just return the exchange filters like this as it will have the same meaning:

```suggestion
   return exchangeFilters;
```

Reply:
Done